model/prelude_mem.sail

```diff
@@ -58,15 +58,18 @@ struct RISCV_strong_access = {
 
 /* The Sail concurrency interface lets us have a physical address type
    with additional information, provided we can supply a function that
-   converts it into a bitvector.  Since we are just using xlenbits as a
-   physical address, we need the identity function for xlenbits. */
-val xlenbits_identity : xlenbits -> xlenbits
+   converts it into a bitvector. When in RV32 mode, physaddrbits is
+   bits(34), which needs to be zero-extended to bits(64) for
+   sail_mem_write/read interface. This conversion is necessary because
+   the memory interface expects 64-bit addresses
```

Comment:
It supports 32 bits too I guess. I'd say "This conversion is necessary because the memory interface only supports 32-bit or 64-bit physical addresses."